File: Java/problem53.java

public class problem53 {

	public static void main(String[] args) {
		
		//System.out.println(nCr(23,10));
		//System.out.println(fac(13) * fac(10));
		//System.out.println(fac(23));
		
	}

	public static long nCr(long n, long r) {
		if(n == r) {
			return 1;
		}
		if(r > n) {
			return 0;
		}
		System.out.println("n: "+n+" n-r: "+ (n-r)+" r; "+r);
		long div = (fac(n-r) * fac(r));
		return fac(n) / div;
	}
	
	public static long fac(long n) {
		if (n == 0) {
			return 1;
		}
		else {
			return n * fac(n-1);
		}
	}
}

File: Python/problem53.py
def fac(n):
    if(n == 0):
        return 1
    return n * fac(n-1)

def nCr(n,r):
    if(r>n):
        return 0;
    return fac(n) / (fac(r) * fac(n-r))


million = 1000000
count = 0

for i in range(1,101):
    for j in range(1,i):
        if(nCr(i,j)>million):
            #print(nCr(i,j))
            count += 1
